src/AzureAuth/ExceptionListToStringConverter.cs

```diff
@@ -51,39 +52,58 @@ public static List<string> ExtractCorrelationIDsFromException(IList<Exception> e
         }
 
         /// <summary>
-        /// Executes the convertion of exceptions to a string.
+        /// Executes the convertion of exceptions to a JSON string.
         /// </summary>
-        /// <param name="exceptions">The exceptions.</param>
-        /// <returns>The <see cref="string"/>.</returns>
+        /// <param name="exceptions">List of exceptions.</param>
+        /// <returns>JSON format of exception list.</returns>
         public static string Execute(IEnumerable<Exception> exceptions)
         {
             if (exceptions == null || exceptions.Count() == 0)
             {
                 return null;
             }
 
-            return string.Join("\n", exceptions.Select(SingleLineException));
+            List<CustomException> customExceptions = new List<CustomException>();
+            foreach (Exception exception in exceptions)
+            {
+                var customException = new CustomException(exception);
+                customExceptions.Add(customException);
+            }
+
+            return JsonSerializer.Serialize(customExceptions);
         }
+    }
 
+    /// <summary>
+    /// CustomException class with only select properties of Exception class.
+    /// </summary>
+    internal class CustomException
+    {
         /// <summary>
-        /// Converts exceptions to a single string.
+        /// Initializes a new instance of the <see cref="CustomException"/> class.
         /// </summary>
-        /// <param name="ex">The exceptions.</param>
-        /// <returns>The <see cref="string"/>.</returns>
-        private static string SingleLineException(Exception ex)
+        /// <param name="exception">Exception.</param>
+        public CustomException(Exception exception)
         {
-            if (ex != null)
+            if (exception?.Message != null)
             {
-                var message = ex.Message
-                                .Replace("\n", string.Empty)
-                                .Replace("\r", string.Empty)
-                                .Replace("\t", string.Empty);
-                return $"{ex.GetType()}: {message}";
+                var singleLineMessage = exception.Message
+                                    .Replace("\n", string.Empty)
+                                    .Replace("\r", string.Empty)
+                                    .Replace("\t", string.Empty);
+                this.Message = $"{exception.GetType()}: {singleLineMessage}";
             }
-            else
+
+            if (exception?.InnerException != null)
             {
-                return "null";
+                this.InnerException = new CustomException(exception.InnerException);
             }
         }
+

```

Comment:
Similarly, it would be helpful if each exception JSON object had an AAD error code if one was present in the message. These are the strings starting with `AADSTS` ([listed here](https://learn.microsoft.com/en-us/azure/active-directory/develop/reference-aadsts-error-codes) ). So we could look pull that out as well and make a property for it if preset. This will make bucketing these MSAL exceptions far far easier in App Insights.